Tests/TensorFlowTests/EpochsTests.swift

```diff
@@ -0,0 +1,307 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+@testable import TensorFlow
+
+var rng = ARC4RandomNumberGenerator(seed: [42])
+
+final class EpochsTests: XCTestCase {
+  func testBaseUse() {
+    // A mock item type that tracks if it was accessed or not
+    class Tracker {
```

Comment:
```suggestion
    class AccessTracker {
```

(or even `ReadTracker`)